src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/BuildHistoryManager.java

```diff
@@ -40,23 +40,26 @@ public BuildHistoryManager(List<Rule> rules) {
      */
     @Override
     public synchronized void perform(Job<?, ?> job) throws IOException, InterruptedException {
+        String uniquePerformName = job.getFullName();
```

Comment:
full name or display name ?

Reply:
full name, else you cannot distinguish two projects with the same display name in different folders